site/docs/starter/slash-commands.md

```diff
@@ -0,0 +1,550 @@
+import {
+  dotEnvConfig,
+  enableCachePlugin, 
+  createBot, 
+  GatewayIntents, 
+  enableCacheSweepers, 
+  startBot 
+} from "./deps.ts";
+import { events } from "./src/events/mod.ts";
+
+const log = logger({ name: "Main" });
+
+// Secrets
+dotEnvConfig({ export: true });
+export const BOT_TOKEN = Deno.env.get("BOT_TOKEN") || "";
+export const BOT_ID = BigInt(atob(BOT_TOKEN.split(".")[0]));
+
+log.info("Starting Bot, this might take a while...");
+
+// Create the Bot Object
+export const bot = enableCachePlugin(createBot({
+  token: BOT_TOKEN,
+  botId: BOT_ID,
+  intents: GatewayIntents.Guilds | GatewayIntents.GuildMembers | GatewayIntents.GuildMessages | GatewayIntents.DirectMessages,
+  events
+}));
+
+enableCacheSweepers(bot);
+
+if (BOT_TOKEN) {
+  // Start the bot
+  await startBot(bot);
+} else {
+  throw "Error: No Token!";
+}
+```
+
+This then exposes `bot.guilds`, `bot.members`, and more. The sweepers will clean the cache to make sure we aren't using
+too much memory.
+
+## Creating a Command Type
+
+Since we are writing this project in TypeScript, we should define a type for our commands to follow. This can be defined
+in `src/commands/mod.ts`; as well as a `Map()` to contain all our commands:
+
+```typescript title="src/commands/mod.ts"
+import { ApplicationCommandOption, ApplicationCommandTypes, Bot, Collection, Interaction } from "../../deps.ts";
+
+// Commands
+import { ping } from "./cmd/ping.ts";
+
+// Subcommands are just commands that can't define other subcommands.
+export type subCommand = Omit<Command, "subcommands">;
+
+export type subCommandGroup = {
+  name: string;
+  subCommands: subCommand[];
+};
+
+export interface Command {
+  name: string;
+  description: string;
+  usage?: string[];
+  options?: ApplicationCommandOption[];
+  type: ApplicationCommandTypes;
+  /** Defaults to `Guild` */
+  scope?: "Global" | "Guild";
+  execute: (bot: Bot, interaction: Interaction) => unknown;
+  subcommands?: Array<subCommandGroup | subCommand>;
+}
+
+export const commands = new Map<string, Command>();
+
+export function addCommand(command: Command) {
+  commands.set(command.name, command);
+  return command;
+};
+
+addCommand(ping);
+```
+
+You may have also noticed a helper function to add a command, this just makes sure that the command's name in the map
+matches it's name in the command map. Anyways, you may have also noticed the definitions of `subCommand` and
+`subCommandGroup`; these allow us to define the structure of sub-commands as well (because discord supports them, so we
+should too).
+
+## Defining Commands
+
+All of our commands will follow the same structure:
+
+```typescript title="src/commands/cmd/ping.ts"
+import { addCommand } from "../mod.ts";
+import { ApplicationCommandTypes } from "../../../deps.ts";
+import { snowflakeToTimestamp, humanizeMilliseconds } from "../../helpers.ts"
+
+export const ping = addCommand({
+  // this command will be used `/ping` in discord.
```

Comment:
I would recommend these comment explanations be done separately below so you can attach images/screenshots of what they look like to the user

Reply:
Good Idea.